src/views/Viewer.vue

```diff
@@ -415,6 +418,19 @@ export default {
 					// retrieve folder list
 					const fileList = await folderRequest(dirPath)
 
+					// retrieve caption languages
```

Comment:
I think this should be moved to the Video component as it is only used for videos

Reply:
Actually i was unable to code the retrieval of the caption files within Videos.vue, because I could not implement it unless I was using an 'async' function and this did not work well with the Videos.vue template -- I got "promises" instead of the the actual list of files.

Please don't hesitate to suggest code snippets... I need guidance :)

Reply:
"Promises" are values JavaScript returns when it does not have the data yet. So when you are getting a file listing, you need to send some requests, if all of the javascript stoped at that moment, that would kill the user experience. Instead, JavaScript returns a promise: "I don'd have this data now, but I promise I will have it". You can also say to JS, ah, OK once you have that data call this function with it as argument (`then`). So what you could do (I think) is:

Add a [created] hook to the Video component. In there you do what you did before to get the data (`yourCodeThatFindsTheRightThingAndReturnsAPromice`) and then use `then` to specify what should happen once we have that data. So in the snippet below `response` will hold the data you want, and 
```
  created() {
      yourCodeThatFindsTheRightThingAndReturnsAPromice().
      .then(response => {
          /* do stuff with the respose to get the right things*/
          this.captions = ...  // ← set a fild to be used in the template on top
      });
  }
  ```

I don't really have that much time to explain it well, but it is really worth the effort to learn about promises and async/await [see this useful tutorial by Mozilla](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises)